Firestore/Swift/Source/PropertyWrapper/FirestoreQueryObservable.swift

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import FirebaseFirestore
+
+@available(iOS 14.0, *)
+@available(tvOS, unavailable)
+internal class FirestoreQueryObservable<T>: ObservableObject {
+  @Published var items: T
+
+  private let firestore = Firestore.firestore()
+  private var listener: ListenerRegistration?
+
+  private var setupListener: (() -> Void)!
+
+  internal var configuration: FirestoreQuery<T>.Configuration {
+    didSet {
+      removeListener()
+      setupListener()
+    }
+  }
+
+  init<U: Decodable>(configuration: FirestoreQuery<T>.Configuration) where T == [U] {
+    items = []
+    self.configuration = configuration
+    setupListener = createListener { [weak self] querySnapshot, error in
+      if let _ = error {
+        // in this scenario, we actively ignore the error
+        self?.items = []
+        return
+      }
+
+      guard let documents = querySnapshot?.documents else {
+        self?.items = []
+        return
+      }
+
+      self?.items = documents.compactMap { document in
```

Comment:
In all three different scenarios, we use `compactMap` to filter out any `nil` results. This is mostly convenience, to relieve developers from having to `compactMap` the results themselves (which would somewhat defat the whole purpose of having a one-stop property wrapper), but do you think we might have situations where users would *want* to see `nil` values in the result set, @morganchen12 @schmidt-sebastian?

Reply:
I could more easily find a use case for preserving nil values for a wrapped value that were a Dictionary from key to decoded entity.
With the exploration in my PR you can basically add overloads for many wrapped types. This could perhaps be used to provide several combinations of values or optional values or result values - in arrays or in dictionaries keyed by the document id - and all of the above wrapped in outer results.
The possibility is there, I just can't say if it's a nice API. 😊

Reply:
> The possibility is there, I just can't say if it's a nice API. 😊

Agree - if the community requests this, it should be easy enough to extend the API accordingly.